### PR TITLE
ci: knip

### PR DESCRIPTION
**Pull request by kanadgupta**
| 🚥 Resolves RM-8001 |
| :------------------- |

## 🧰 Changes

Brings in [`knip`](https://github.com/webpro/knip) to help us identify code/package excess.

## 🧬 QA & Testing

Does everything pass?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "build": "npx lerna run build --stream",
     "build-docs": "node bin/build-markdown.js",
     "clean": "npx lerna clean",
-    "lint": "npm run lint:types; npm run lint:js; npm run prettier",
+    "lint": "knip && npm run lint:types && npm run lint:js && npm run prettier",
```
Comment (kanadgupta) [inline package.json:9]: the commands before the semicolons weren't throwing properly failed 😬

File: packages/httpsnippet-client-api/test/index.test.ts [hunk 2]
```diff
@@ -95,15 +94,6 @@ describe('httpsnippet-client-api', () => {
       let consoleStub;
 
       beforeEach(async () => {
-        try {
```
Comment (erunion) [inline packages/httpsnippet-client-api/test/index.test.ts:98]: Did Knip find this or you?

**Review by erunion — APPROVED** [review]

File: packages/httpsnippet-client-api/test/index.test.ts (hunk 2 above)
Comment (kanadgupta) [reply packages/httpsnippet-client-api/test/index.test.ts:98]: it flagged rimraf so sort of!

Comment (erunion) [reply packages/httpsnippet-client-api/test/index.test.ts:98]: dang

**Pull request merged**